src/main/java/com/hivemq/extensions/handler/PluginInitializerHandler.java

```diff
@@ -173,15 +175,15 @@ private void fireInitialize(
             @Override
             public void onSuccess(@Nullable final Void result) {
                 authenticateWill(ctx, msg, promise);
-                ctx.channel().attr(ChannelAttributes.CONNECT_MESSAGE).set(null);
+                removeHandler(ctx);
             }
 
             @Override
             public void onFailure(final @NotNull Throwable t) {
                 Exceptions.rethrowError(t);
                 log.error("Calling initializer failed", t);
                 ctx.channel().attr(ChannelAttributes.CONNECT_MESSAGE).set(null);
-                ctx.writeAndFlush(msg, promise);
+                removeHandler(ctx);
```

Comment:
I think the handler can directly be removed after calling fireInitialize in line 110.